### ISSUE TITLE
More accurate way of determining connection status.

### ISSUE THREAD

**Issue by SoneeJohn** [duanefields/VirtualKVM#31]
Sometimes the client may indicated that it's connected even if it's not in TDM. I noticed that this comes times happens if the thunderbolt cable is pulled in but TDM was not enabled.

**Comment by SoneeJohn** [comment]
I think I found a solution:

```

#pragma mark - Target Display Mode Status

- (BOOL)clientIsInTargetDisplayMode {
    
    NSArray *screens = [NSScreen screens];
    
    if (screens.count == 0) {
        return NO;
    }
    
    NSMutableArray <NSNumber *> *screenNumbers = [NSMutableArray new];
    for (NSScreen *screen in screens) {
        
        NSLog(@"Screen name: %@",screen.deviceDescription[@"NSScreenNumber"]);
        
        if (screen.deviceDescription[@"NSScreenNumber"]) {
            
            [screenNumbers addObject:@([screen.deviceDescription[@"NSScreenNumber"] unsignedIntValue])];
        }
    }
    
    if (screenNumbers.count == 0) {
        return NO;
    }
    
    NSMutableArray <NSString *> *localizedScreenNames = [NSMutableArray new];
   
    for (NSNumber *screenNumber in screenNumbers) {
        
        NSString *localizedScreenName = [self screenNameForDisplay:screenNumber.unsignedIntValue];
        if (localizedScreenName && localizedScreenName.length != 0) {
            [localizedScreenNames addObject:localizedScreenName];
            [localizedScreenNames addObject:@"iMac"];
        }
        
    }
    
    if (localizedScreenNames.count == 0) {
        return NO;
    }
    
    for (NSString *localizedScreenName in localizedScreenNames) {
        
        if ([localizedScreenName isEqualToString:@"iMac"]) {
            return YES;
            break;
        }
    }
    
    return NO;
}

- (NSString *)screenNameForDisplay:(CGDirectDisplayID)displayID {
    
    NSString *screenName = nil;
    
    NSDictionary *deviceInfo = (__bridge NSDictionary *)IODisplayCreateInfoDictionary(CGDisplayIOServicePort(displayID), kIODisplayOnlyPreferredName);
    NSDictionary *localizedNames = [deviceInfo objectForKey:[NSString stringWithUTF8String:kDisplayProductName]];
    
    if ([localizedNames count] > 0) {
        screenName = [localizedNames objectForKey:[[localizedNames allKeys] objectAtIndex:0]];
    }
    
    return screenName;
}

```

**Comment by SoneeJohn** [comment]
See https://github.com/SoneeJohn/VirtualKVM/tree/accurate-connection-status

**Comment by SoneeJohn** [comment]
Closed with #32
